Logs step function article (#7)
MaximeVivier/Articles: blog-posts/step-function-learnings/step-function-logs-learning.md

---
published: false

> title: "Stop looking for your express state machine logs, they don't exist"

Reviewer: I think a title more in line with the google search that people will do when facing this exact problem will be more impactful

Reviewer: Adding AWS somewhere would be nice

> description: 'Logs for express Step Function with the CDK'

Reviewer: ```suggestion
description: 'How to have logs for express Step Function with the CDK'
```

Reviewer: Idem, adding AWS somewhere would be good, like `AWS Step Functions`

> tags: logs, stepfunction, serverless, cdk

Reviewer: You can use tags with more impact, see https://dev.to/tags to find them

series:
canonical_url:
---

> In this article you will learn how to add **logs in an express step function** with the CDK because it is not enabled by default. When you go from a standard step function with logs to an express one that gives you nothing.

Reviewer: Add `AWS step function` aswell.
The second sentence is a bit odd. You can reformulate with something along the line: `Since it is enabled by default on the Standard version, ...`
You should add multiple links to the documentation, ie. one for "it is not enabled by default" with this link: https://docs.aws.amazon.com/step-functions/latest/dg/diff-standard-express-exec-details-ui.html#exp-wf-exec-limitation-cw-reliance-test

> TL;DR

Reviewer: The TL;DR should be a single phrase that explains the solution with the configuration that needs to be added.
Here you are explaining everything, before revealing the solution. This is misleading.
Moreover, you should put it as a header

## Standard workflows have logs enabled by default inside AWS Step Function
> :orange_book: Read this to better grasp what is happening with those log groups in Step Function or skip to the next section for the tutorial you came for

> AWS explains it really clearly in its documentation, here is a quote describing the default logs behaviors of both Standard and Express types of State Machine.

Reviewer: Add a link

>Standard Workflows record execution history in AWS Step Functions, although you can optionally configure logging to Amazon CloudWatch Logs.
>
>Unlike Standard Workflows, Express Workflows don't record execution history in AWS Step Functions. To see execution history and results for an Express Workflow, you must configure logging to Amazon CloudWatch Logs. Publishing logs doesn't block or slow down executions.

> In the following images, you can see that for three different State Machines one Express configured with logs, one Express configured without logs and one Standard, the only one that has a log group associated is the Express one configured with a log group.

Reviewer: ```suggestion
The following images show three different State Machines: one Express configured with logs, one Express configured without logs and one Standard. The only one that has a log group associated is the Express one configured with a log group.
```

![Three types of state machine](./assets/three-types-of-state-machine.png 'Three types of state machine')

![One log group associated to one of the Express State machine](./assets/one-log-group.png 'One log group associated to one of the Express State machine')

And the logs available for the Standard type of State Machine are inside the AWS Step Function service but there is no log group associated.

![Logs of Standard State Machine is in the AWS Step Function service](./assets/one-log-group.png 'Logs of Standard State Machine is in the AWS Step Function service')

## :computer: CDK is your best friend

![CDK is da real MVP](./assets/youDaRealMVP.jpeg 'CDK is da real MVP')

> All you need is to create a log group, configure it and then attach it to the State Machine via the logs prop. If you want to jump directly to the snippet [here](#white_check_mark-how-to-configure-logs-in-an-express-step-function-in-cdk) is a shortcut.

Reviewer: The anchor does not work on the preview: [here](https://dev.to/maximevivier/dum-title-62p-temp-slug-4063551?preview=56dc3b8ff89339a45b1ce06e26ee44a543b9d7a81703cd55ff237f723cad9f2005620921d700970ef1e69b2ee12520fb59734a630cc6af07271f213e#white_check_mark-how-to-configure-logs-in-an-express-step-function-in-cdk)

> The cdk handles all the work of creating the role for the step function to write inside CloudWatch.

Reviewer: ```suggestion
The CDK handles all the work of creating the role for the step function to write inside CloudWatch.
```

> The LogLevel allows you to select what kind of information you want to have. The four levels are: `OFF`, `ALL`, `ERROR` and `FATAL`.

Reviewer: You should recommend using `ALL`, as described in https://docs.aws.amazon.com/step-functions/latest/dg/diff-standard-express-exec-details-ui.html#exp-wf-exec-limitation-details-log-dependent-test
